src/main/java/io/github/sashirestela/cleverclient/CleverClient.java

```diff
@@ -41,16 +47,23 @@ public class CleverClient {
      *                    server sent events (SSE). Optional.
      */
     @Builder
-    public CleverClient(@NonNull String urlBase, @Singular List<String> headers, HttpClient httpClient,
+    public CleverClient(String baseUrl, String urlBase, @Singular List<String> headers, HttpClient httpClient,
             String endOfStream) {
-        this.urlBase = urlBase;
+        if ((baseUrl == null || baseUrl.isEmpty()) && (urlBase == null || urlBase.isEmpty())) {
```

Comment:
Please use [CommonUtil.isNullOrEmpty ](https://github.com/sashirestela/cleverclient/blob/main/src/main/java/io/github/sashirestela/cleverclient/util/CommonUtil.java#L16) for these validations.